packages/core/src/rules/common/hide-internals.ts

```diff
@@ -0,0 +1,95 @@
+import { Oas3Rule, Oas2Rule } from '../../visitors';
```

Comment:
In general, we've put decorators into the rules folder and I think they should go into a decorators folder instead.

Reply:
```suggestion
import { Oas3Decorator, Oas2Decorator } from '../../visitors';
```
Also, I think these would be the correct types...